codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -496,19 +520,49 @@ private void incrementPendingBytes(int numBytes) {
              */
             int write(int allowedBytes) {
                 int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
+                int writtenBytes = 0;
                 try {
-                    connectionState().incrementStreamWindow(-writtenBytes);
-                    incrementStreamWindow(-writtenBytes);
+                    boolean writeError = true;
+                    try {
+                        writing = true;
```

Comment:
@Scottmitch I think your suggestion for the changes to the `FlowControlled` interface would effectively boil down to the same problem as the `writing` boolean.  I suggest we just throw if `writing = true` to avoid re-entrance.  That seems reasonable to me ... WDYT?